### PR TITLE
killing the silly nonstatic layout system

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
Special Size Measurements detected layout systems, `layoutSystemForChildren` and `parentLayoutSystem`. The options were `'flex' | 'grid' | 'flow'` and a rather badly named `'nonfixed'` which was meant to basically mean "pinSystem". This was totally wrong for two reasons: 1. it should have been called "nonstatic" and it should have detected whether the position is static of an element or not 2. even if something is "nonfixed" it still actually makes a flow layout.

**Fix:**
Kill nonfixed layout system type. Added a new boolean flag called `providesBoundsForChildren`. If providesBoundsForChildren is true AND the layoutSystem is `flow`, it means it is a "pinned system", ie you can expect to use pins on the children and they will work properly.

**Commit Details:**
- remove `nonfixed` from layout system options
- rename and fix behavior of `isElementNonFixed` to `isElementNonStatic` which is now used to tell if the element provides a coordinate system for its children.
- fixed the Inspector's "Layout System" section which was unable to show Pins as selected for months now. Now it shows Pins if the detected layout system is flow && `providesBoundsForChildren === true`

### CODE REVIEW

**Review by enidemi — APPROVED** [review]

File: editor/src/components/editor/actions/actions.ts [hunk 1]
```diff
@@ -686,7 +686,7 @@ function switchAndUpdateFrames(
       case 'flex':
         return 'flex'
       case LayoutSystem.PinSystem:
-        return 'nonfixed'
+        return 'flow'
```
Comment (seanparsons) [inline editor/src/components/editor/actions/actions.ts:689]: This makes my eye twitch! :)

File: editor/src/components/canvas/dom-walker.ts [hunk 2]
```diff
@@ -47,22 +47,19 @@ function elementLayoutSystem(computedStyle: CSSStyleDeclaration | null): Detecte
...
   return positionValues.includes(valueAsAny) ? valueAsAny : null
 }
 
-function isElementNonFixed(computedStyle: CSSStyleDeclaration | null) {
+function isElementNonStatic(computedStyle: CSSStyleDeclaration | null) {
   if (computedStyle == null) {
     return false
   }
-  if (computedStyle.position != null && computedStyle.position !== 'fixed') {
+  if (computedStyle.position != null && computedStyle.position !== 'static') {
```
Comment (Rheeseyb) [inline editor/src/components/canvas/dom-walker.ts:62]: Seems we don't need the `!= null` checks at the start of these

**Review by seanparsons — APPROVED** [review]
As long as the tests are fixed, looks good to me!

**Comment by Rheeseyb** [comment]
> As long as the tests are fixed, looks good to me!

Looking at the failing tests it seems like there is a bug in the logic and not the tests?

**Comment by seanparsons** [comment]
> Looking at the failing tests it seems like there is a bug in the logic and not the tests?

That's what I meant, wasn't saying the tests themselves were specifically wrong.

**Comment by balazsbajorics** [comment]
I forgot about this PR! also, the tests!

**Comment by balazsbajorics** [comment]
turns out the tests were previously incorrect, the new snapshot is the correct one!

**Pull request merged**